multiversx_sdk_cli/tests/test_wallet_generator.py

```diff
@@ -37,3 +41,36 @@ def test_derive_pem_from_mnemonic(monkeypatch: Any):
         assert True
     else:
         assert False
+
+
+def test_generate_wallet_and_save_in_json_format(monkeypatch: Any):
+    output_path = Path(__file__).parent / "testdata-out" / "testWallet.json"
+
+    with monkeypatch.context() as mp:
+        mp.setattr('getpass.getpass', lambda _: "TestPassword")
+
+        result = main(['wallet', 'new', '--json', '--output-path', str(output_path)])
+    
+    assert Path.is_file(output_path) == True
+
+    if not result:
```

Comment:
As previously, the assert statements can be written a bit differently - e.g. more concise.